<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1000</width>
    <height>1000</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>Parser</string>
  </property>
  <property name="windowIcon">
   <iconset>
    <normaloff>logo.png</normaloff>logo.png</iconset>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QWidget" name="mainWidget" native="true">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>1000</width>
      <height>971</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: rgb(255, 255, 255);</string>
    </property>
    <widget class="QLabel" name="insertCodeLabel">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>20</y>
       <width>341</width>
       <height>41</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">color: rgb(6, 96, 159);
font: 75 16pt &quot;Georgia&quot;;</string>
     </property>
     <property name="text">
      <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:14pt;&quot;&gt;Insert your code here:&lt;br/&gt;&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
     </property>
    </widget>
    <widget class="QTextEdit" name="inputCodeTextEdit">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>50</y>
       <width>471</width>
       <height>151</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">border-color: rgb(4, 72, 117);
font: 10pt &quot;DejaVu Sans Mono&quot;;
background-color: rgb(212, 232, 247);</string>
     </property>
     <property name="placeholderText">
      <string notr="true"/>
     </property>
    </widget>
    <widget class="QPushButton" name="parseCodePushButton">
     <property name="geometry">
      <rect>
       <x>190</x>
       <y>210</y>
       <width>161</width>
       <height>61</height>
      </rect>
     </property>
     <property name="cursor">
      <cursorShape>PointingHandCursor</cursorShape>
     </property>
     <property name="styleSheet">
      <string notr="true">color: rgb(255, 255, 255);
font: 75 16pt &quot;Cambria&quot;;
background-color:rgb(7, 119, 197);
border-radius:30px;
</string>
     </property>
     <property name="text">
      <string>Parse Code</string>
     </property>
    </widget>
    <widget class="QWidget" name="parsingResultGroup" native="true">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>280</y>
       <width>981</width>
       <height>681</height>
      </rect>
     </property>
     <widget class="QWebEngineView" name="parseTreeWebEngineView" native="true">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>50</y>
        <width>971</width>
        <height>621</height>
       </rect>
      </property>
     </widget>
     <widget class="Line" name="line">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>0</y>
        <width>971</width>
        <height>20</height>
       </rect>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
     <widget class="QLabel" name="parseTreeLabel">
      <property name="geometry">
       <rect>
        <x>360</x>
        <y>20</y>
        <width>271</width>
        <height>31</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">color: rgb(6, 96, 159);
font: 75 16pt &quot;Georgia&quot;;</string>
      </property>
      <property name="text">
       <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p align=&quot;center&quot;&gt;Parse Tree:&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="invalidSyntaxGroup" native="true">
     <property name="geometry">
      <rect>
       <x>540</x>
       <y>90</y>
       <width>441</width>
       <height>151</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true"/>
     </property>
     <widget class="QLabel" name="errorIcon">
      <property name="geometry">
       <rect>
        <x>80</x>
        <y>10</y>
        <width>41</width>
        <height>51</height>
       </rect>
      </property>
      <property name="text">
       <string/>
      </property>
      <property name="pixmap">
       <pixmap>res/error.webp</pixmap>
      </property>
      <property name="scaledContents">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QLabel" name="syntaxErrorLabel">
      <property name="geometry">
       <rect>
        <x>140</x>
        <y>10</y>
        <width>181</width>
        <height>41</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">font: 24pt &quot;Haettenschweiler&quot;;
color: rgb(255, 0, 4);</string>
      </property>
      <property name="text">
       <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Syntax Error&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
      </property>
     </widget>
     <widget class="QLabel" name="errorMessageLabel">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>70</y>
        <width>441</width>
        <height>61</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">font: 10pt &quot;DejaVu Sans Mono&quot;;
color: rgb(255, 0, 4);
</string>
      </property>
      <property name="text">
       <string>Error Message</string>
      </property>
      <property name="wordWrap">
       <bool>true</bool>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="validSyntaxGroup" native="true">
     <property name="geometry">
      <rect>
       <x>620</x>
       <y>90</y>
       <width>251</width>
       <height>71</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true"/>
     </property>
     <widget class="QLabel" name="checkIcon">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>51</width>
        <height>61</height>
       </rect>
      </property>
      <property name="text">
       <string/>
      </property>
      <property name="pixmap">
       <pixmap>res/check.png</pixmap>
      </property>
      <property name="scaledContents">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QLabel" name="validSyntaxLabel">
      <property name="geometry">
       <rect>
        <x>60</x>
        <y>10</y>
        <width>181</width>
        <height>41</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">font: 24pt &quot;Haettenschweiler&quot;;
color: rgb(46, 204, 113);</string>
      </property>
      <property name="text">
       <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Valid Syntax&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
      </property>
     </widget>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1000</width>
     <height>26</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuShow">
    <property name="title">
     <string>Show</string>
    </property>
    <addaction name="showParsingTableAction"/>
    <addaction name="showDemoAction"/>
   </widget>
   <widget class="QMenu" name="menuDeveloped_by">
    <property name="title">
     <string>Developed by</string>
    </property>
    <addaction name="actionYoussef_Geroge_19P9880"/>
    <addaction name="actionMostafa_Nasrat_19P4619"/>
    <addaction name="actionAnthony_Amgad_19P9880"/>
    <addaction name="actionKerrolos_Wageeh_19P3468"/>
   </widget>
   <addaction name="menuShow"/>
   <addaction name="menuDeveloped_by"/>
  </widget>
  <action name="actionYoussef_Geroge_19P9880">
   <property name="text">
    <string>Youssef George    19P9824</string>
   </property>
  </action>
  <action name="actionMostafa_Nasrat_19P4619">
   <property name="text">
    <string>Mostafa Nasrat    19P4619</string>
   </property>
  </action>
  <action name="actionAnthony_Amgad_19P9880">
   <property name="text">
    <string>Anthony Amgad  19P9880</string>
   </property>
  </action>
  <action name="actionKerrolos_Wageeh_19P3468">
   <property name="text">
    <string>Kerrolos Wageeh 19P3468</string>
   </property>
  </action>
  <action name="showParsingTableAction">
   <property name="icon">
    <iconset>
     <normaloff>parsingtable.png</normaloff>parsingtable.png</iconset>
   </property>
   <property name="text">
    <string>Parsing Table</string>
   </property>
   <property name="statusTip">
    <string>Opens SLR(1) parsing table</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+T</string>
   </property>
  </action>
  <action name="showDemoAction">
   <property name="text">
    <string>Demo</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QWebEngineView</class>
   <extends>QWidget</extends>
   <header>qwebengineview.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="res.qrc"/>
 </resources>
 <connections/>
</ui>
